services/memory_storage/memory_storage.cpp

```diff
@@ -256,19 +274,82 @@ namespace services {
                   session.data());
             return;
         }
+        auto collection_names = logical_plan->collection_dependencies();
+        context_storage_t collection_dependencies;
+        for (const auto& name : collection_names) {
+            if (!check_collection_(session, name)) {
+                trace(log_,
+                      "memory_storage_t:execute_plan_ collection not found {}, sesion: {}",
+                      name.to_string(),
+                      session.data());
+                return;
+            }
+            collection_dependencies.emplace(name, collections_.at(name).context);
+        }
         sessions_.emplace(session, session_t{logical_plan, current_message()->sender(), 1});
-        actor_zeta::send(collections_.at(logical_plan->collection_full()),
+        auto plan =
+            collection::planner::create_plan(collection_dependencies, logical_plan, components::ql::limit_t::unlimit());
```

Comment:
@OrangeSmokingJacket 
Right now I'm working on index features and I have to place some new information inside physical plan like `mdisk_` and `sessions_` how can do this now? any hints?

Reply:
Add a case for operator_type inside collection::execute_sub_plan, maybe? This section is for collection specifics. It will work for mdisk_.
The same could be done in memory storage in memory_storage_t::execute_plan, for memory_storage stuff